pkg/cluster/install.go

```diff
@@ -300,11 +300,14 @@ func setFieldCreatedByHive(createdByHive bool) database.OpenShiftClusterDocument
 func (m *manager) bootstrap() []steps.Step {
 	s := []steps.Step{}
 
+	// initialize required client to manage cluster credentials, for both CSP or WI clusters
 	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
 		s = append(s,
 			steps.Action(m.ensureClusterMsiCertificate),
 			steps.Action(m.initializeClusterMsiClients),
 		)
+	} else {
+		s = append(s, steps.Action(m.initializeClusterSPClients))
```

Comment:
What's the idea for moving the initialize part for ClusterSP Clients here before the validateResources?
The other 2 functions `ensureClusterMsiCertificate` and `initializeClusterMsiClients` were needed to be before the validation as the clientIDs and principalIDs for usamsi and platformidentities would be needed for the validation.
Also, for the same reason the IDs populating should be done before validateResources.

Reply:
This was mainly done to keep consistency between these equivalent steps for CSP vs WI management wherever possible. I'll reorder the steps for both CSP and WI to what you said.

Reply:
Pushed up a fix for this, lmk what you think.

Reply:
Yes, looks good to me. I think it's good from consistency point of view, it will not break anything, I was just concerned to play with CSP authorizer before the dynamic validations.